extensions/wikia/PhalanxII/models/PhalanxContentModel.php

```diff
@@ -18,11 +18,13 @@ public function __construct( Title $title ) {
 	 *
 	 * @return bool
 	 */
-	public function isOk() {
+	public function isOk(): bool {
+		global $wgUser;
+
 		return (
```

Comment:
Here it is worth to mention one thing. Consider the following code:
```php
function one() {
    echo "One!\n";
    return true;
}

function two() {
    echo "Two!\n";
    return false;
}

function three() {
    echo "Three!\n";
    // Some very expensive logic here
    // Which we do not need to execute every time.
    return true;
}

$result = one() || two() || three();
// Will only output:
// One!
```
Getting back to your code: which expression is most likely to be `true` and which expression is the most expensive to evaluate?

Considering https://github.com/Wikia/app/blob/f18c868bd1a0b4a972d30a520172b1e82fcd63da/extensions/wikia/PhalanxII/models/PhalanxContentModel.php#L11 - unless we deliberately modify `$this->title` after `$this` is instantiated, the part `!( $this->title instanceof Title )` will always be `false` as `$this->title` will always be a `Title`.

Then, `$this->isWikiaInternalRequest()` operates on `$this->wg->Request` whereas `global $wgUser` creates a new reference. I suspect the latter is slower and less probable than the former one.

This might not be of top importance here, it's just a nice opportunity for some general considerations about Boolean algebra and the order of execution. :-)